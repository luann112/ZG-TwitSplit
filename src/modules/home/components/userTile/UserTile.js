import React, { Component } from 'react';
import Router from 'next/router';
import {
  UserTileWrapperStyled,
  UserNameStyled
} from './UserTile.styles';
import Avatar from 'src/components/avatar';

class UserTile extends Component {
  toChatPage = () => {
    const { initUser, user } = this.props;
    Router.push('/chat');
    if (initUser && typeof initUser === 'function' && user.id && user.name) {
      initUser(user.id, user.name);
    }
  }
  render() {
    const { user } = this.props;
    return (
      <UserTileWrapperStyled onClick={this.toChatPage}>
        <Avatar src={user.avatar} />
        <UserNameStyled>{user.name}</UserNameStyled>
      </UserTileWrapperStyled>
    );
  }
}

export default UserTile;