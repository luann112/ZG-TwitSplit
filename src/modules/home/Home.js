import React, { Component } from 'react';
import { connect } from 'react-redux';
import { actions } from 'src/reducers/auth';
import {
  HomeWrapperStyled,
  TextStyled,
  UsersWrapperStyled,
} from './Home.styles';
import { SAMPLE_USERS } from 'lib/enums';
import UserTile from './components/userTile';

class Home extends Component {

  renderUser = (user) => user.id !== 'broadcast' && (
    <UserTile 
      key={user.id}
      user={user}
      initUser={this.props.initUser}
    />
  );
  
  render() {
    return (
      <HomeWrapperStyled className="container">
        <TextStyled>Who are you?</TextStyled>
        <UsersWrapperStyled>
          {SAMPLE_USERS.map(this.renderUser)}
        </UsersWrapperStyled>
      </HomeWrapperStyled>
    );
  }
}

const mapStateToProps = () => ({});
const mapDispatchToProps = {
  initUser: actions.initUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);