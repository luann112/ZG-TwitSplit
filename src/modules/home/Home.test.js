import { shallow } from 'enzyme';
import Home from './Home';
import UserTile from './components/userTile';

describe('<Home />', () => {
  const wrapper = shallow(<Home />);
  it('should render', () => {
    expect(wrapper.length).toBe(1);
  });
  it('should match to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
