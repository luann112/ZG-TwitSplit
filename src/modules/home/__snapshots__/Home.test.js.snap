// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match to snapshot 1`] = `ShallowWrapper {}`;
